### PR TITLE
Fix the IP address retrieval error

### PR DESCRIPTION
**Pull request by xiaheshun**
// Describe your PR here; eg. Fixes #issueNo

Make sure that:

-------------------------------------------test code----------------------------------------------------------
The original code: 

  if (!ip.isLoopbackAddress()){
	System.out.print(ip.getHostAddress());
	System.out.print("-------");
	System.out.print(netInterface.getDisplayName());
	System.out.print("-------");
	System.out.print(netInterface.getName());
	System.out.println();
}

fe80:0:0:0:3507:b35:d4e4:ebf3%eth2-------Bluetooth Device (Personal Area Network)-------eth2
fe80:0:0:0:7881:df0f:e380:188b%wlan2-------Microsoft Wi-Fi Direct Virtual Adapter-------wlan2
172.23.39.81-------Intel(R) Ethernet Connection (3) I218-LM-------eth3
fe80:0:0:0:58b0:4f02:d55e:4f32%eth3-------Intel(R) Ethernet Connection (3) I218-LM-------eth3
fe80:0:0:0:8010:123e:b12b:1e57%wlan3-------Intel(R) Dual Band Wireless-AC 7265-------wlan3
192.168.137.1-------Microsoft Wi-Fi Direct Virtual Adapter #2-------wlan4
fe80:0:0:0:e495:d3f:9590:e7d7%wlan4-------Microsoft Wi-Fi Direct Virtual Adapter #2-------wlan4
192.168.16.1-------Hyper-V Virtual Ethernet Adapter-------eth22
fe80:0:0:0:9d08:73ac:5e28:4f53%eth22-------Hyper-V Virtual Ethernet Adapter-------eth22
------------------------------------------test code-----------------------------------------------------------

These are the network cards for my laptop.   The shenyu utility class only gets IP 192.168.16.1, but my real IP is 172.23.39.81.

I had fix the IP address retrieval error.

Use the new rule:

example:   eth3  172.23.39.81   |    eth22  192.168.16.1 |  wlan4  192.168.137.1

I will spit as "name namePostfix  ipv4Postfix"
 "eth  3  81", "eth  22  1", "wlan  4  1"

then, sort the list:
   first rule:          name                 (other name<"enp"<"eth"<"bond")
   second rule:     namePostfix       ( 3[eth] > 22[eth] > 4[wlan])
   third rule :        ipv4Postfix         ( 81[172.23.39.81] > 1[192.168.16.1] > 1[192.168.137.1] )

then, keep old filter (ipv4).

- [ √ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor).
- [ √ ] You submit test cases (unit or integration tests) that back your changes.
- [ √ ] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-common/src/main/java/org/apache/shenyu/common/utils/IpUtils.java [hunk 1]
```diff
@@ -17,9 +17,10 @@
 
 package org.apache.shenyu.common.utils;
 
-import java.net.InetAddress;
-import java.net.NetworkInterface;
-import java.util.Enumeration;
+import java.io.Serializable;
+import java.net.*;
```
Comment (yu199195) [inline shenyu-common/src/main/java/org/apache/shenyu/common/utils/IpUtils.java:21]: do not use import *

File: shenyu-common/src/main/java/org/apache/shenyu/common/utils/IpUtils.java [hunk 2]
```diff
@@ -51,45 +57,112 @@ public static String getHost() {
...
+                        }
+                    }
+                } else {
+                    hostIp = ipv4Result.get(0).getIp();
+                }
+            } else if (!ipv6Result.isEmpty()) {
+                hostIp = ipv6Result.get(0).getIp();
             }
+        } catch (SocketException e) {
+            e.printStackTrace();
```
Comment (yu199195) [inline shenyu-common/src/main/java/org/apache/shenyu/common/utils/IpUtils.java:153]: not used e.printStackTrace();

File: shenyu-common/src/main/java/org/apache/shenyu/common/utils/IpUtils.java [hunk 3]
```diff
@@ -136,4 +209,87 @@ private static boolean ipMatch(final String ip, final String pattern) {
...
         return dp[m][n];
     }
+
+    /**
+     * To obtain a prefix.
+     *
+     * @param name
+     * @return the name
+     */
+    private static String getName(String name) {
```
Comment (yu199195) [inline shenyu-common/src/main/java/org/apache/shenyu/common/utils/IpUtils.java:219]: add finnal  , such as  :  finnal String name

**Comment by tydhot** [comment]
why not add some test about the util either? I think it can help us review too.

File: shenyu-common/src/main/java/org/apache/shenyu/common/utils/IpUtils.java [hunk 4]
```diff
@@ -136,4 +210,87 @@ private static boolean ipMatch(final String ip, final String pattern) {
...
+    private static class NetCard implements Serializable {
+        private String ip;
+        private String name;
+        private Integer namePostfix;
+        private Integer ipv4Postfix;
+
+        public NetCard() {
+        }
+
+        public NetCard(String ip, String name, Integer namePostfix) {
```
Comment (yu199195) [inline shenyu-common/src/main/java/org/apache/shenyu/common/utils/IpUtils.java:251]: add finnal

File: shenyu-common/src/main/java/org/apache/shenyu/common/utils/IpUtils.java [hunk 5]
```diff
@@ -136,4 +210,87 @@ private static boolean ipMatch(final String ip, final String pattern) {
...
+            this.ip = ip;
+            this.name = name;
+            this.namePostfix = namePostfix;
+        }
+
+        public NetCard(String ip, String name, Integer namePostfix, Integer postfix) {
```
Comment (yu199195) [inline shenyu-common/src/main/java/org/apache/shenyu/common/utils/IpUtils.java:257]: add finnal

File: shenyu-common/src/main/java/org/apache/shenyu/common/utils/IpUtils.java [hunk 6]
```diff
@@ -136,4 +210,87 @@ private static boolean ipMatch(final String ip, final String pattern) {
...
+
+        public Integer getIpv4Postfix() {
+            return ipv4Postfix;
+        }
+
+        public Integer getNamePostfix() {
+            return namePostfix;
+        }
+
+        public void setNamePostfix(Integer namePostfix) {
```
Comment (yu199195) [inline shenyu-common/src/main/java/org/apache/shenyu/common/utils/IpUtils.java:288]: add finnal

**Comment by yu199195** [comment]
pls add finnal to all Method parameters

**Review by yu199195 — APPROVED** [review]

**Comment by yu199195** [comment]
@xiaheshun  welcome to join apache shenyu controbutor 。 you can add my wechat： xixy199195

**Pull request merged**